addons/delivery/models/stock_move.py

```diff
@@ -19,8 +19,7 @@ def _cal_move_weight(self):
 
     def _get_new_picking_values(self):
         vals = super(StockMove, self)._get_new_picking_values()
-        carrier_id = self.group_id.sale_id.carrier_id.id
-        vals['carrier_id'] = self.rule_id.propagate_carrier and carrier_id
+        vals['carrier_id'] = self.mapped('rule_id.propagate_carrier') and self.mapped('group_id.sale_id.carrier_id').id
```

Comment:
Pay attention that `mapped` returns a list and `False` is not the same as `[False]` as you can see on this little test 
![image](https://user-images.githubusercontent.com/12071695/129708480-82022154-d38a-47cd-a114-d808e635fb71.png)

I would rather test there is at least one `True` in self.mapped('rule_id.propagate_carrier')

Reply:
```suggestion
        carrier_ids = self.mapped('group_id.sale_id.carrier_id')
        carrier_id = len(carrier_ids) == 1 and carrier_ids or False
        vals['carrier_id'] = all(self.mapped('rule_id.propagate_carrier')) and carrier_id.id
```